docs/api/guidelines.md

```diff
@@ -0,0 +1,342 @@
+developers to more easily understand how your API works, and also makes it easier to implement a client which 
+works with your API.
+
+There are a few custom methods which tend to crop up from time to time, including:
+  * Methods that are part of other protocols (e.g. Oauth2)
+  * Long running, complex tasks
+  * Bulk operations
+  * Search queries
+
+We don't currently have any long running (batch), bulk operations, or search methods as part of our API, but may add them
+in the future. This document will be updated with additional requirements and guidelines.
+
+Custom methods should use the HTTP ***POST*** method call. 
+
+## Return Payloads
+
+### Field Naming
+
+Naming fields, similar to naming resources, can be tricky. Use standard naming fields whenever possible. Some
+examples include:
+
+| Name           | Format             | Description                      |
+| :---           | :-----             | :----------                      |
+| created        | RFC 3339 timestamp | Time a resource was created      |
+| expires        | RFC 3339 timestamp | Time a resource expires          |
+| id             | string             | UID for a resource               |
+| lastSeenAt     | RFC 3339 timestamp | Time a resource was last seen    |
+| name           | string             | Name of a resource               |
+| privilege      | string             | Role or permission               |
+| resource       | string             | Resource name                    |
+| subject        | string             | UID for a user or group          |
+| updated        | RFC 3339 timestamp | Time a resource was last updated |
+
+If you are creating a compound field name, use lower camel case for the JSON field with the first character
+lowercase, and capitalize the first letter of each subsequent word (e.g. thisIsAField). Do not use characters
+other that ASCII a-z and A-Z inside of the field name.
+
+### Sequential IDs
+
+When returning a resource, avoid using a sequential ID in the ID field. Using sequential IDs allows
+callers to potentially guess a particular resource which can lead to leaking information. We
+have a Polymorphic ID in our `infrahq/infra/api` package which you should use instead.
+
+### Payload Structure
+
+When you are returning a payload, keep the following things in mind:
+  * Keep things simple
+  * Avoid unnecessary deep nested structures inside of resource payloads
+  * If you nest one resource into another resource, make certain that it maintains the same structure as if the resource
+    were on its own
+
+## Timestamps
+
+For timestamp fields, we follow [RFC 3339](https://datatracker.ietf.org/doc/html/rfc3339) which overlaps
+with ISO 8601. Timestamps are always stored in in UTC form. The client is responsible for converting any
+timestamp from UTC to local time.
+
+Timestamp fields follow the form:
+
+```YYYY-MM-DDThh:mm:ss.SSSZ```
```

Comment:
Looks like this renders ok, but is confusing githubs syntax highlighting.

```suggestion
`YYYY-MM-DDThh:mm:ss.SSSZ`
```

Or maybe add newlines before/after the ```

Reply:
I'll switch this to single ticks.